### ISSUE TITLE
Allow for caption to be specified for Table components

### ISSUE THREAD

**Issue by chawes13** [LaunchPadLab/lp-components#463]
We should be able to specify the text _or_ component to use as a [`<caption>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/caption) element in our Table components. We could leverage a similar pattern as established for label / labelComponent in [LabeledField](https://github.com/LaunchPadLab/lp-components/blob/0ea9404056f8bece2fd3fe84e09de57222a7c985/src/forms/labels/labeled-field.js#L96), i.e., accept a `caption` string prop and a `captionComponent` prop that defaults to `({ caption }) => <caption>{caption}</caption>`.

This is related to research by @mwislek in addressing #324

**Comment by dpikt** [comment]
👍good callout. I think we could just make this a node prop that will be inserted into the tag since it wouldn't need access to any "internal" props/state (I think):

```
caption="My Table"

...

caption={<p>My<strong>important</strong>table</p>}
```

**Comment by chawes13** [comment]
Is there a pattern for accepting `Node|String`? Passing in text seems like the primary use case, and `caption={<>My Table</>}` feels a little clunky

**Comment by dpikt** [comment]
A string is a node, so it would work that way by default 🙂 My example above was an "inclusive or" of the ways it could be used.

**Comment by dpikt** [comment]
To be clear, I mean node as in `PropTypes.node`, not a DOM node.

**Comment by chawes13** [comment]
Awesome, thanks for the clarification!
